lib/src/metta/runner/stdlib.rs

```diff
@@ -1012,12 +1026,215 @@ impl Grounded for MatchOp {
     }
 }
 
+#[derive(Clone, PartialEq, Debug)]
+pub struct UniqueOp {
+    pub(crate) space: DynSpace,
+}
+
+impl UniqueOp {
+    pub fn new(space: DynSpace) -> Self {
+        Self{ space }
+    }
+}
+
+impl Display for UniqueOp {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "unique")
+    }
+}
+
+impl Grounded for UniqueOp {
+    fn type_(&self) -> Atom {
+        Atom::expr([ARROW_SYMBOL, ATOM_TYPE_ATOM, ATOM_TYPE_ATOM])
+    }
+
+    fn execute(&self, args: &[Atom]) -> Result<Vec<Atom>, ExecError> {
+        let arg_error = || ExecError::from("unique expects single executable atom as an argument");
+        let atom = args.get(0).ok_or_else(arg_error)?;
+
+        // TODO: Calling interpreter inside the operation is not too good
```

Comment:
Indeed.  There are two possible solutions here, but both are beyond the scope of this PR.

1.) Make the interpreter aware of which arguments need to be interpreted / reduced, so the reduction could happen before this `execute` method is invoked. or
2.) Make this method able to essentially act lazily.  Same pattern as async Rust, which returns a future object, and is evaluated by a runtime.

My vote would be for 1, because 2 comes with a lot of overhead.

Reply:
Actually there is a third way which is possible with Minimal MeTTa interpreter:

3. one can return a minimal MeTTa `chain` which reduces the argument first and then call second part of the operation which manipulates the evaluated value (probably it is what Adam's comment says). It is the reason why `interpret_no_error` function is inside `non_minimal_only_stdlib` module.

But for this case specifically it is enough to change the `unique` op type to the `(-> %Undefined% Atom)` and argument is evaluated before call is made.

Reply:
For example `UniqueOp` could have the following implementation:
```rs
impl Grounded for UniqueOp {
    fn type_(&self) -> Atom {
        Atom::expr([ARROW_SYMBOL, ATOM_TYPE_UNDEFINED, ATOM_TYPE_ATOM])
    }

    fn execute(&self, args: &[Atom]) -> Result<Vec<Atom>, ExecError> {
        let arg_error = || ExecError::from("unique expects single executable atom as an argument");
        let atom = args.get(0).ok_or_else(arg_error)?;

        let mut expr: ExpressionAtom = atom.clone().try_into()?;
        let mut set = GroundingSpace::new();
        expr.children_mut().retain(|x| {
            let not_contained = set.query(x).is_empty();
            if not_contained { set.add(x.clone()) };
            not_contained
        });
        Ok(expr.into_children())
    }

    fn match_(&self, other: &Atom) -> MatchResultIter {
        match_by_equality(self, other)
    }
}
```